### DISCUSSION TITLE
Assembly repo not recognized

### DISCUSSION THREAD

**Discussion by ckrause** [github-linguist/linguist#6212] [Classification] [answered]
Linguist does not recognize any content in this repo: [loda-lang/loda-programs](https://github.com/loda-lang/loda-programs). It contains more than 100,000 `*.asm` files but they are not classified as Assembly. Individual files are recognized correctly, but not for the entire repo. To my knowledge, it should not be filter out as vendor, generated or documentation.

Interestingly, for an earlier version of the repo, it works!: [ckrause/loda](https://github.com/ckrause/loda)

Is there any way to enable debug logs or to find out why it is not recognized?

**Comment by lildude** [comment, marked as answer]
> It contains more than 100,000 `*.asm` files

There's your problem. For performance reasons, Linguist will not analyse any repo with more than 100000 files:

https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/repository.rb#L132

**Reply by ckrause** [reply]
Thanks for the explanation. Is there any way to override this and manually set the repo type to Assembly?

**Reply by lildude** [reply]
Unfortunately not as there’s no such thing as a repo type. The “type” you see is really showing the most popular language in the repo. Your only option to give any sort of searchable context to the content is to use topics.
